libvast/src/system/start_command.cpp

```diff
@@ -61,6 +61,9 @@ caf::message start_command_impl(start_command_extra_steps extra_steps,
   if (!parsers::endpoint(str, node_endpoint))
     return caf::make_message(
       make_error(ec::parse_error, "invalid endpoint", str));
+  // Default to port 42000/tcp if none is set.
```

Comment:
I would add a note that if you pass `0` to CAF, the expected behavior is that it will select a random high-level port. In fact, I would probably put a log message on VERBOSE to let the user know that we're going to select a random port.

Reply:
Setting 0 explicitly makes VAST use 42000 with this change.

Reply:
Shouldn't the logic be as follows?
- if nothing is specified, default to 42000
- if 0 is specified, use 0

Reply:
That doesn't really make sense, port 0 is not a valid port.

Reply:
Specifying `0` means the operating system chooses a free port for you, I think we should not prevent that.

Reply:
TIL actually, did not know that. Will take care of it accordingly.

Reply:
I've implemented this accordingly.